src/overlays/actors/ovl_En_Test3/z_en_test3.c

```diff
@@ -27,113 +180,1123 @@ const ActorInit En_Test3_InitVars = {
+        this->unk_D78 = &D_80A41854[6];
+        this->player.actor.textId = this->unk_D78->textId;
+        this->player.actor.flags |= (ACTOR_FLAG_1 | ACTOR_FLAG_8);
+    }
+}
+
+void func_80A409D4(EnTest3* this, PlayState* play) {
+    if ((play->actorCtx.flags & ACTORCTX_FLAG_5) || (play->actorCtx.flags & ACTORCTX_FLAG_4)) {
+        play->actorCtx.flags &= ~ACTORCTX_FLAG_4;
+        func_80A3F0B0(this, play);
+        ActorCutscene_SetReturnCamera(CAM_ID_MAIN);
+    } else {
+        sEnTest3_Input = *CONTROLLER1(&play->state);
+    }
+}
+
+void func_80A40A6C(EnTest3* this, PlayState* play) {
+    gSaveContext.save.weekEventReg[64] |= 0x20;
+}
+
+void EnTest3_Update(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    EnTest3* this = THIS;
+
+    sEnTest3_Input.rel.button = sEnTest3_Input.cur.button;
+    sEnTest3_Input.cur.button = 0;
+    sEnTest3_Input.rel.stick_x = 0;
+    sEnTest3_Input.rel.stick_y = 0;
+
+    play->actorCtx.flags &= ~ACTORCTX_FLAG_7;
+    this->player.actor.draw = EnTest3_Draw;
+    D_80A41D48 = false;
+    this->player.actor.flags &= ~(ACTOR_FLAG_1 | ACTOR_FLAG_8);
+    if (Cutscene_CheckActorAction(play, 0x1FA) &&
+        !((this->player.actor.category == ACTORCAT_PLAYER) &&
+          ((play->actorCtx.flags & ACTORCTX_FLAG_5) || (play->actorCtx.flags & ACTORCTX_FLAG_4)))) {
+        if (this->player.csMode != 5) {
+            play->startPlayerCutscene(play, &this->player, 5);
+        }
+        play->actorCtx.flags &= ~ACTORCTX_FLAG_4;
+    } else if (this->player.actor.category == ACTORCAT_PLAYER) {
+        func_80A409D4(this, play);
+    } else if (play->startPlayerCutscene(play, &this->player, 0)) {
+        if (this->unk_D88 >= 7) {
+            Vec3f worldPos;
+
+            Math_Vec3f_Copy(&worldPos, &this->player.actor.world.pos);
+            this->unk_D80 = 4;
+            func_80A40230(this, play);
+            Math_Vec3f_Copy(&this->player.actor.world.pos, &worldPos);
+            D_80A41D48 = false;
+            this->unk_D84 = 0.0f;
+        }
+    } else {
+        D_80A41D40 = 0.0f;
+        D_80A41D44 = this->player.actor.shape.rot.y;
+        this->unk_D94(this, play);
+        sEnTest3_Input.press.button =
+            (sEnTest3_Input.rel.button ^ sEnTest3_Input.cur.button) & sEnTest3_Input.cur.button;
+        func_800B6F20(play, &sEnTest3_Input, D_80A41D40, D_80A41D44);
+    }
+    play->playerUpdate(&this->player, play, &sEnTest3_Input);
+    if (D_80A41D48) {
+        this->player.actor.world.pos.x = D_80A41D50.x;
+        this->player.actor.world.pos.z = D_80A41D50.z;
+        this->player.linearVelocity = 0.0f;
+    }
+}
+
+s32 D_80A418C8 = false;
+
+s32 EnTest3_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnTest3* this = THIS;
+
+    if (limbIndex == OBJECT_TEST3_LIMB_01) {
+        D_80A41D6C = &this->player.meleeWeaponInfo[2].base;
```

Comment:
This is `sPlayerCurBodyPartPos` on the player_lib branch, it is supposed to be a negative access to bodyPartPos. You don't have to name this variable if you don't want to, but fix the struct access please
```suggestion
        D_80A41D6C = &this->player.bodyPartsPos[-1];
```
More info at:
https://github.com/Decompollaborate/mm/blob/cc167bcda9d57fa90b4dcb6a4a97b02a0bf8023c/src/code/z_player_lib.c#L2002-L2007

Reply:
Huh, that's really weird.